workflow/sync/sync_manager.go

```diff
@@ -11,20 +12,52 @@ import (
 
 type NextWorkflow func(string)
 type GetSyncLimit func(string) (int, error)
+type IsWorkflowDeleted func(string) bool
 
 type Manager struct {
 	syncLockMap  map[string]Semaphore
 	lock         *sync.Mutex
 	nextWorkflow NextWorkflow
 	getSyncLimit GetSyncLimit
+	isWFDeleted  IsWorkflowDeleted
 }
 
-func NewLockManager(getSyncLimit GetSyncLimit, nextWorkflow NextWorkflow) *Manager {
+func NewLockManager(getSyncLimit GetSyncLimit, nextWorkflow NextWorkflow, isWFDeleted IsWorkflowDeleted) *Manager {
 	return &Manager{
 		syncLockMap:  make(map[string]Semaphore),
 		lock:         &sync.Mutex{},
 		nextWorkflow: nextWorkflow,
 		getSyncLimit: getSyncLimit,
+		isWFDeleted:  isWFDeleted,
+	}
+}
+
+func (cm *Manager) getWorkflowKey(key string) (string, error) {
+	if key == "" {
+		return "", fmt.Errorf("holderkey is empty")
+	}
+	items := strings.Split(key, "/")
```

Comment:
this seems redundant - you just split and then re-join - I think you can always assume namespace/name

Reply:
In template level synchronization scenario, Key will have 3 items (including node name)